specs/admin_spec.rb

```diff
@@ -0,0 +1,233 @@
+require_relative 'spec_helper'
+
+describe "Admin class" do
+  describe "Initializer" do
+    it "is an instance of Admin" do
+      admin = Hotel::Admin.new
+
+      admin.must_be_kind_of Hotel::Admin
+    end
+
+    it "is set up for specific attributes and data types" do
+      admin = Hotel::Admin.new
+
+      admin.must_respond_to :rooms
+      admin.rooms.length.must_equal 20
+      admin.must_respond_to :reservations
+      admin.reservations.must_be_kind_of Array
+    end
+  end
+
+  describe "book_room method" do
+    it "raises error if input is not instance of reservation" do
+      admin = Hotel::Admin.new
+      reservation = Hotel::Reservation.new(['12 May', '13 May'])
+
+      proc {admin.book_room("")}.must_raise ArgumentError
+      proc {admin.book_room('12 May 2017')}.must_raise ArgumentError
+    end
+
+    it "removes a room from rooms list" do
+      admin = Hotel::Admin.new
+      res1 = Hotel::Reservation.new(['12 May', '13 May'])
+      free_rooms = admin.rooms.length
+
+      admin.book_room(res1)
+
+      admin.rooms.length.must_equal free_rooms - 1
```

Comment:
This test doesn't make sense, the number of rooms doesn't change when a room is booked, the number of free rooms in a date range would change.